src/Composer/ComposerWidget.vala

```diff
@@ -490,7 +490,26 @@ public class Mail.ComposerWidget : Gtk.Grid {
     }
 
     private void on_discard () {
-        discarded ();
+        var discard_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("Delete This Draft?"), // choice of 'Delete' over 'Discard' for clarity
+            _("Are you sure you want to delete this draft permanently?"),
+            "dialog-error",
```

Comment:
Since this isn't an error message `dialog-error` is probably not appropriate here
```suggestion
            "dialog-warning",
```